### DISCUSSION TITLE
Using Keras and pytorch with make_reduction

### DISCUSSION THREAD

**Discussion by HishamSalem** [alan-turing-institute/sktime#2566] [Q&A]
Hello, 

I was wondering if there is a way to utilize sktime's make_reduction with the estimator being a model from keras/pytorch?

if there is another function that does that, I would love to know.

**Comment by fkiraly** [comment]
@HishamSalem, that should be perfectly possible, although the estimator needs to conform to the sklearn regressor interface (for supervised regressors) or to the sktime regressor interface (for time series regressors).

For `keras`, you should be able to do this with `keras.wrappers.scikit_learn.KerasRegressor`, i.e., wrap with that first before passing to `make_regressor`. Would be interested to hear about your results, e.g., whether it works well with reduction and pipelines.

For `pytorch`, you can use `skorch` https://github.com/skorch-dev/skorch, that will conform to the sklearn interface.

**Reply by HishamSalem** [reply]
@fkiraly , It does work well. A challenge however would have to be how we shape the data in terms of inputs. For example, it works for an MLP however when working with CNN and LSTMS it ends up being more interesting since we have to reshape the layers from the inputs. 

I did try it with GRUs but I get a recursion error.

**Reply by fkiraly** [reply]
hm, interesting - did not see this reply yet, @HishamSalem, apologies.

FYI @AurumnPegasus

**Comment by philip-ndikum** [comment]
Any updates on this?

**Reply by fkiraly** [reply]
it should be possible out of the box, as long as the estimators conform with the `sklearn` interface.

Regarding @HishamSalem's issue, I can't really say what was going on there due to lack of code - I was hoping for a bug report.

As said, as long as the estimator is `sklearn` compatible, you should be able to stick it into `make_reduction`.
